src/ipc.ts

```diff
@@ -0,0 +1,204 @@
+
+	return promise;
+}
+
+// async function findEndpoint(tries = 0): Promise<string> {
+// 	if (tries > 30) {
+// 		throw new Error('Could not find endpoint');
+// 	}
+
+// 	const endpoint = `http://127.0.0.1:${6_463 + (tries % 10)}`;
+
+// 	try {
+// 		const response = await fetch(endpoint);
+
+// 		if (response.status === 404) return endpoint;
+
+// 		return await findEndpoint(tries + 1);
+// 	} catch {
+// 		return findEndpoint(tries + 1);
+// 	}
+// }
+
+export function encode(op: number, data: RPCMessagePayload | string | {}) {
+	const stringifiedData = JSON.stringify(data);
+	const length = Buffer.byteLength(stringifiedData);
+	const packet = Buffer.alloc(8 + length);
+
+	packet.writeInt32LE(op, 0);
+	packet.writeInt32LE(length, 4);
+	packet.write(stringifiedData, 8, length);
+
+	return packet;
+}
+
+interface WorkingData {
+	full: string;
+	op: number | undefined;
+}
+
+const working: WorkingData = {
+	full: '',
+	op: undefined,
+};
+
+export function decode(socket: Socket, callback: (data: { op: OPCodes; data: RPCMessagePayload }) => void) {
+	const packet = socket.read();
+	if (!packet) {
+		return;
+	}
+
+	let { op } = working;
+	let raw;
+	if (working.full === '') {
+		op = working.op = packet.readInt32LE(0);
+		const len = packet.readInt32LE(4);
+		raw = packet.slice(8, len + 8);
+	} else {
+		raw = packet.toString();
+	}
+
+	try {
+		const data = JSON.parse(working.full + raw);
+		callback({ op: op!, data });
+		working.full = '';
+		working.op = undefined;
+	} catch {
+		working.full += raw;
+	}
+
+	decode(socket, callback);
+}
+
+export class IPCTransport extends AsyncEventEmitter {
+	private socket: Socket | null;
+
+	public constructor(public readonly client: RPCClient) {
+		super();
+
+		this.socket = null;
+	}
+
+	public async connect() {
+		this.socket = await getIPC();
+
+		const socket = this.socket;
+
+		socket.on('close', this.onClose.bind(this));
+		socket.on('error', this.onClose.bind(this));
+
+		this.emit('open');
+
+		socket.write(
+			encode(OPCodes.Handshake, {
+				v: 1,
+				client_id: this.client.clientId,
+			}),
+		);
+
+		socket.pause();
+
+		socket.on('readable', () => {
+			decode(socket, ({ op, data }) => {
+				switch (op) {
+					case OPCodes.Ping:
+						this.send(data, OPCodes.Pong);
+						break;
+					case OPCodes.Frame:
+						if (!data) {
+							return;
+						}
+
+						// if (data.cmd === RPCCommands.Authorize && !('evt' in data)) {
+						// 	findEndpoint()
+						// 		.then((endpoint) => {
+						// 			this.client.request.endpoint = endpoint;
+						// 		})
+						// 		.catch((error) => {
+						// 			this.client.emit('error', error);
+						// 		});
+						// }
+
+						this.emit('message', data);
+						break;
+					case OPCodes.Close:
+						this.emit('close', data);
+						break;
+					default:
+						break;
+				}
+			});
+		});
+	}
+
+	public onClose(error: boolean) {
+		this.emit('close', error);
+	}
+
+	public send(data: RPCMessagePayload | string | {}, op = OPCodes.Frame) {
```

Comment:
`{}` here is redundant. Did you mean to type this as an empty object? If so, use `Record<string, never>`. `{}` represents any value except `null` and `undefined`.